packages/integration-tests/tests/bridge_mainnet_tokens.ts

```diff
@@ -0,0 +1,768 @@
+import type { BridgeFungibleToken } from '@fuel-bridge/fungible-token';
+import {
+  RATE_LIMIT_AMOUNT,
+  RATE_LIMIT_DURATION,
+} from '@fuel-bridge/solidity-contracts/protocol/constants';
+import {
+  CustomToken,
+  CustomTokenWETH,
+  CustomToken__factory,
+  CustomTokenWETH__factory,
+} from '@fuel-bridge/solidity-contracts/typechain';
+import {
+  USDT_ADDRESS,
+  USDC_ADDRESS,
+  WBTC_ADDRESS,
+  WETH_ADDRESS,
+} from '@fuel-bridge/solidity-contracts/protocol/constants';
+// import set_canonical_token_bytecode from '@fuel-bridge/solidity-contracts/deploy/hardhat/helpers/set_canonical_token_bytecode';
+import type { TestEnvironment } from '@fuel-bridge/test-utils';
+import {
+  setupEnvironment,
+  relayCommonMessage,
+  waitForMessage,
+  createRelayMessageParams,
+  getOrDeployL2Bridge,
+  FUEL_TX_PARAMS,
+  getMessageOutReceipt,
+  fuel_to_eth_address,
+  waitForBlockCommit,
+  waitForBlockFinalization,
+  getTokenId,
+  getBlock,
+  FUEL_CALL_TX_PARAMS,
+  hardhatSkipTime,
+} from '@fuel-bridge/test-utils';
+import chai from 'chai';
+import { toBeHex, parseEther } from 'ethers';
+import type { JsonRpcProvider, Signer } from 'ethers';
+import { Address, BN } from 'fuels';
+import type {
+  AbstractAddress,
+  WalletUnlocked as FuelWallet,
+  MessageProof,
+} from 'fuels';
+
+const { expect } = chai;
+
+const tokenAddresses: string[] = [
+  USDT_ADDRESS,
+  USDC_ADDRESS,
+  WBTC_ADDRESS,
+  WETH_ADDRESS,
+];
+
+const decimals: bigint[] = [6n, 6n, 8n];
+
+for (const [index, value] of tokenAddresses.entries()) {
```

Comment:
I think you can cut the test time if you move this loop lower, so that you don't need to be deploying contracts in each loop, you deploy just once